src/core/sys/posix/dirent.d

```diff
@@ -276,8 +302,20 @@ else version( FreeBSD )
     void   seekdir(DIR*, c_long);
     c_long telldir(DIR*);
 }
-else version (OSX)
+else version ( OSX )
 {
+    version ( D_LP64 )
+    {
+        pragma(mangle, "seekdir$INODE64") void seekdir(DIR*, c_long);
+        pragma(mangle, "telldir$INODE64") c_long telldir(DIR*);
+    }
+    else
+    {
+        // 32-bit mangles __DARWIN_UNIX03 specific functions with $UNIX2003 to
+        // maintain backward compatibility with binaries build pre 10.5
+        pragma(mangle, "seekdir$INODE64$UNIX2003") void seekdir(DIR*, c_long);
```

Comment:
Hi Martin!  Yes, that is the way it is prototyped in direct.h, and the return for `telldir` is also a C long, so just a 32-bit directory stream location when compiled with -m32.   Prototype in dirent.h is:

```C
void seekdir(DIR *, long) __DARWIN_ALIAS_I(seekdir);
```

where `__DARWIN_ALIAS_I` from cdef.h mangles specifically for 32-bit arch using 64-bit inodes (`seekdir$INODE64$UNIX2003`)

It can be confirmed without have macosx SDK installed by viewing https://opensource.apple.com/source/Libc/Libc-1044.1.2/include/dirent.h.